lib/Runtime/Base/Entropy.cpp

```diff
@@ -64,6 +71,21 @@ void Entropy::AddIoCounters()
         Add((char *)&ioc.WriteTransferCount,  sizeof(ioc.WriteTransferCount));
         Add((char *)&ioc.OtherTransferCount,  sizeof(ioc.OtherTransferCount));
     }
+#elif defined(__LINUX__)
+    // It's difficult to implement GetProcessIoCounters on Linux in a
+    // satisfactory way. We'll just use /dev/random to generate an
+    // equivalent amount of random bytes (6 times 64 bytes).
+#define ENTROPY_BYTES 64 * 6
+    int ret;
+    char buffer[ENTROPY_BYTES];
+    size_t size = sizeof(buffer);
+
+    ret = syscall(SYS_getrandom, buffer, size, 0);
+    if (ret == size)
+    {
+        Add(buffer, size);
```

Comment:
What do you think if we can use the implementation under pal/src/misc/random.cpp? We might hold a bit bigger cache there though.